packages/good-design/src/core/buttons/ClaimButton.tsx

```diff
@@ -214,6 +213,9 @@ const ClaimButton = ({
         showActionModal();
         await handleClaim();
         setClaimLoading(true);
+        // Here we assume claiming was completed succesfully,
+        // so it pushes the default route without params to prevent repeated claim requests
+        window.history.pushState(null, "", window.location.origin + window.location.hash);
```

Comment:
it's a **dirty** workaround

first-of-all we're making our code native base e.g. cross-platform. this fixes web only 
second thing - react apps have own navigation state management so any library shouldn't manipulate it **directly**

**this should be reverted**

do such redirect at protocol UI side (via nav library used there)

Reply:
Its not doing a redirect, above just clears history without refresh
But if we need to then yes, we can do it

Reply:
@L03TJ3 this shouldn't be here, and shouldn't be done through raw API.

Reply:
pls fix ASAP

Reply:
fixed, don't refresh but just don't execute when claimed